src/Microsoft.ML.Data/TrainCatalog.cs

```diff
@@ -102,14 +103,34 @@ internal CrossValidationResult(ITransformer model, T metrics, IDataView scores,
             foreach (var split in DataOperationsCatalog.CrossValidationSplit(Environment, data, splitColumn, numFolds))
             {
                 var model = estimator.Fit(split.TrainSet);
-                var scoredTest = model.Transform(split.TestSet);
+                IDataView scoredTest;
+
+                if (IsCastableToTransformerChainOfITransformer(model))
+                    scoredTest = (Unsafe.As<TransformerChain<ITransformer>>(model)).Transform(split.TestSet, TransformerScope.Everything);
+                else
+                    scoredTest = model.Transform(split.TestSet);
                 result[fold] = new CrossValidationResult(model, scoredTest, fold);
                 fold++;
             }
 
             return result;
         }
 
+        private static bool IsCastableToTransformerChainOfITransformer(object o)
+        {
+            var type = o.GetType();
+            while (!type!.FullName!.StartsWith("Microsoft.ML.Data.TransformerChain`1[", StringComparison.Ordinal))
```

Comment:
>while [](http://example.com/codeflow?start=12&length=5)

do you think it is worth checking the assembly name too when finding the type?

Reply:
I don't think so. I don't think they will even get to this point if they had the wrong assembly.